### PR TITLE
SCons: Remove code for unsupported compilers

### PR DESCRIPTION
**Pull request by ghost**
Follow up to: #99217 and #98842

### CODE REVIEW

**Comment by dustdfg** [comment]
There is also one place where `VCINSTALLDIR` is used. I don't know much about VS so have a question. It is defined for all VS versions or only for VS >= 2017?

https://github.com/godotengine/godot/blob/c264100da2d021b3d107f18f5adf74bdaeb60547/platform/windows/detect.py#L51-L52

File: platform/windows/detect.py [hunk 1]
```diff
@@ -112,25 +112,6 @@ def detect_build_env_arch():
...
-                "BIN;": "x86_32",
-            }
-            for path, arch in path_arch.items():
-                final_path = VCINSTALLDIR + path
-                if final_path in PATH:
-                    return arch
-
         # VS 2017 and newer.
         if os.getenv("VCTOOLSINSTALLDIR"):
             host_path_index = os.getenv("PATH").upper().find(os.getenv("VCTOOLSINSTALLDIR").upper() + "BIN\\HOST")
```
Comment (Repiteo) [inline platform/windows/detect.py:117]: This conditional can be removed & its contents bumped down
```suggestion
        host_path_index = os.getenv("PATH").upper().find(os.getenv("VCTOOLSINSTALLDIR").upper() + "BIN\\HOST")
        if host_path_index > -1:
            first_path_arch = os.getenv("PATH")[host_path_index:].split(";")[0].rsplit("\\", 1)[-1].lower()
            if first_path_arch in msvc_target_aliases.keys():
                return msvc_target_aliases[first_path_arch]
```

**Comment by dustdfg** [comment]
> Oop, should've specified, the fix I showed extended to the rest of the conditional; you'll have to remove lines 120-123 as well

Already was checking for correctness. Pressing "commit changes" and then checking to pr's head is the easiest way to get the changes in local which I know... Btw thanks for clarifying! :slightly_smiling_face:

**Comment by Repiteo** [comment]
It also seems that `VCINSTALLDIR` indeed *is* on later VS versions, or at least on my VS2022 version. I assume then that the later check is more thorough and/or `VCTOOLSINSTALLDIR` wasn't introduced until VS2017.

**Comment by dustdfg** [comment]
> It also seems that `VCINSTALLDIR` indeed _is_ on later VS versions, or at least on my VS2022 version. I assume then that the later check is more thorough and/or `VCTOOLSINSTALLDIR` wasn't introduced until VS2017.

Looks like I understood words but honestly I couldn't parse it. Looks like I've already got debuff on mind capabilities :sweat_smile: Could you rephrase please?

**Comment by dustdfg** [comment]
It is not so much sense two merge deletion of two lines of stale code which won't create any problems. I will glue it with some other misc edits
